### ISSUE TITLE
Rule proposal: No duplicate destructuring

### ISSUE THREAD

**Issue by nickofthyme** [sindresorhus/eslint-plugin-unicorn#759]
It would be nice to have a rule that combines/simplifies a duplicated object destructuring. Similar to [`no-duplicates`](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-duplicates.md).

### Bad
```ts
const { defaultColor } = module;
const { transparentColor } = module;
const { defaultD3Color } = module;
```

```ts
const { defaultColor } = module;
let { transparentColor } = module;
const { defaultD3Color } = module;
```

### Good
```ts
const { defaultColor, transparentColor, defaultD3Color } = module;
```

```ts
const { defaultColor, defaultD3Color } = module;
let { transparentColor } = module;
```

**Comment by papb** [comment]
I think this should be merged with #248

**Comment by nickofthyme** [comment]
closing in favor of combining with #248
